tiledb/sm/query/reader_base.cc

```diff
@@ -133,6 +136,37 @@ void ReaderBase::compute_result_space_tiles(
 /*        PROTECTED METHODS       */
 /* ****************************** */
 
+Status ReaderBase::evolve_fragment_array_schemas() {
+  if (fragment_metadata_.size() <= 1) {
+    // No need to evolve
+    return Status::Ok();
+  }
+  for (auto i = fragment_metadata_.size() - 1; i > 0; i--) {
+    if (fragment_metadata_[i]->array_schema_name() ==
```

Comment:
Comparing subsequent fragments will not guarantee they have the same schema name. It might be better to store a set to compare against.